Dagens Nyheter.js

```diff
@@ -0,0 +1,230 @@
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2018 Sebastian Berlin
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+// attr()/text() v2
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+
+
+function detectWeb(doc, url) {
+	if (url.includes('/nyheter/')) {
+		return "newspaperArticle";
+	} else if (url.includes('/sok/') && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.search-item > a');
+	for (let i=0; i<rows.length; i++) {
+		let href = rows[i].href;
+		if (href.includes("/recept/")) {
+			// Exclude recepies.
+			continue;
+		}
+		let title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (let i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+
+	translator.setHandler('itemDone', function (obj, item) {
+		item.publicationTitle = "Dagens Nyheter";
+
+		var nameString = attr(doc, 'div[class="js-article"]', "data-authors");
+		var author = ZU.cleanAuthor(nameString, "author");
+		if(author.firstName === "") {
+			// If there's only one name, the author is not a person,
+			// e.g. "TT".
+			author.firstName = undefined;
+			author.fieldMode = true;
+		}
+		item.creators = [author];
+
+		var abstractString =
+			attr(doc, 'div[class="js-article"]', "data-article-description");
+		item.abstractNote = abstractString.replace("&nbsp;", " ");
```

Comment:
Also, both here and for section below, please test for the scraped string before processing it, since a missing string will otherwise cause an error, i.e. simply `if (abstractString) item.abstractNote .= ..`